src/java/api.js

```diff
@@ -24,8 +24,33 @@ module.exports = class JavaRealmAPI extends RealmAPI {
     return await this.rest.put(`/worlds/${realmId}/${state}`)
   }
 
+  async deleteRealm (realmId) {
+    return await this.rest.delete(`/worlds/${realmId}`)
+  }
+
   async getRealmWorldDownload (realmId, slotId) {
     const data = await this.rest.get(`/worlds/${realmId}/slot/${slotId}/download`)
     return new Download(this, data)
   }
+
+  async getRealmSubscriptionInfo (realmId) {
```

Comment:
If endpoints match on both Java and Bedrock the method to call that endpoint can be declared once in the `RealmAPI` class